papermill/cli.py

```diff
@@ -29,8 +33,8 @@ def print_papermill_version(ctx, param, value):
 
 
 @click.command(context_settings=dict(help_option_names=['-h', '--help']))
-@click.argument('notebook_path')
-@click.argument('output_path')
+@click.argument('notebook_path', required=False)
+@click.argument('output_path', required=False)
```

Comment:
The only awkward thing here is that it's required if not piped, yet there's likely not a way to specify this for click. It's like we want to require it yet override it during one of the cases below (somehow before?).

Reply:
You can do the `INPUT_PIPED = S_ISFIFO(os.fstat(0).st_mode)` and `OUTPUT_PIPED = sys.stdout.isatty()` outside the function set `required` based on those variables is probably the best tradeoff.

Reply:
I agree. Do you want something like the below? It seems to work in every case, but for a detail: when an input pipe exists, the two arguments become optional.
```
import os
import sys
from stat import S_ISFIFO
import click

INPUT_PIPED = S_ISFIFO(os.fstat(0).st_mode)
OUTPUT_PIPED = sys.stdout.isatty()


@click.command(context_settings=dict(help_option_names=['-h', '--help']))
@click.argument('notebook_path', required=not INPUT_PIPED)
@click.argument('output_path', required=not (INPUT_PIPED or OUTPUT_PIPED))
def papermill(notebook_path, output_path):
    if INPUT_PIPED and notebook_path and not output_path:
        output_path = notebook_path
        notebook_path = '-'
    else:
        notebook_path = notebook_path or '-'
        output_path = output_path or '-'

    print('notebook_path={}'.format(notebook_path))
    print('output_path={}'.format(output_path))


if __name__ == '__main__':
    papermill()
```

Reply:
Yeah that about what I was expecting. And I think the logic for required there is correct to match expected behavior.

On a related note we'll want to test this on a variety of OS's despite those methods not having explicit warnings. I can check the windows / linux setups easily enough.